blueocean-dashboard/src/main/js/util/serverBrowserTimeHarmonize.js

```diff
@@ -0,0 +1,64 @@
+import moment from 'moment';
+import { logging } from '@jenkins-cd/blueocean-core-js';
+
+const logger = logging.logger('io.jenkins.blueocean.dashboard.harmonizeTimes');
+/**
+ * we need to make sure that we calculate with the correct time offset
+ *
+ * @param run
+ * @param skewMillis
+ * @returns {{ durationMillis: diff, endTime: Date, startTime: Date}}
+ */
+export class TimeManager {
+    currentTime() {
+        return moment();
+    }
+
+    harmonizeTimes(run, skewMillis) {
+        logger.debug('skewMillis', skewMillis);
+        if (!run.startTime) {
+            logger.error('not found any startTime, seems that a component should not have called this me');
+            return {};
+        }
+    // What time is it now on the client
+        const clientTime = this.currentTime();
+    // what is the start time of the server
+        const serverStartTime = moment(run.startTime);
+    // sync server start date to local time via the skewMillis
+        if (skewMillis < 0) {
+            serverStartTime.add({ milliseconds: Math.abs(skewMillis) });
+        } else {
+            serverStartTime.subtract({ milliseconds: skewMillis });
+        }
+    // export the harmonized start time
+        const startTime = serverStartTime.toJSON();
+    // how long has passed from the start to now in milliseconds
+        const timeElapsed = clientTime.diff(serverStartTime, 'milliseconds');
+    // assume we do not have an end date
+        let endTime = null;
```

Comment:
Actually it is a combination of IDE formatting and then afterwards using lint:fix. Seems comments are not being intended like normal code.